File: dox2dox_app/lib/data/models/organization.dart
import '../../core/enums/app_enums.dart';
import 'user.dart';

class Organization {
  final String id;
  final String name;
  final String? address;
  final String? city;
  final String? state;
  final String? practiceType;
  final String inviteCode;
  final OrgStatus status;
  final String? reviewNotes;
  final DateTime? verifiedAt;
  final DateTime createdAt;
  final int memberCount;

  const Organization({
    required this.id,
    required this.name,
    required this.address,
    required this.city,
    required this.state,
    required this.practiceType,
    required this.inviteCode,
    required this.status,
    required this.reviewNotes,
    required this.verifiedAt,
    required this.createdAt,
    required this.memberCount,
  });

  factory Organization.fromJson(Map<String, dynamic> j) => Organization(
        id: j['id'] as String,
        name: j['name'] as String,
        address: j['address'] as String?,
        city: j['city'] as String?,
        state: j['state'] as String?,
        practiceType: j['practice_type'] as String?,
        inviteCode: j['invite_code'] as String,
        status: OrgStatus.fromWire(j['status'] as String),
        reviewNotes: j['review_notes'] as String?,
        verifiedAt: j['verified_at'] != null ? DateTime.parse(j['verified_at'] as String) : null,
        createdAt: DateTime.parse(j['created_at'] as String),
        memberCount: (j['member_count'] ?? 0) as int,
      );
}

class OrgMember {
  final User user;
  final OrgRole orgRole;
  final DateTime joinedAt;
  final PresenceStatus presence;

  const OrgMember({
    required this.user,
    required this.orgRole,
    required this.joinedAt,
    required this.presence,
  });

  factory OrgMember.fromJson(Map<String, dynamic> j) => OrgMember(
        user: User.fromJson(j['user'] as Map<String, dynamic>),
        orgRole: OrgRole.fromWire(j['org_role'] as String),
        joinedAt: DateTime.parse(j['joined_at'] as String),
        presence: PresenceStatus.fromWire(j['presence'] as String?),
      );
}

File: doqto_backend/app/schemas/organization.py
from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, Field

from app.core.enums import (
    DirectoryVisibility,
    ExternalDmPolicy,
    OrgRole,
    OrgStatus,
    PracticeType,
)
from app.schemas.common import ORMModel


class OrgCreateIn(BaseModel):
    name: str = Field(min_length=1, max_length=255)
    address: str | None = None
    city: str | None = None
    state: str | None = Field(default=None, min_length=2, max_length=2)
    practice_type: PracticeType | None = None


class OrgJoinIn(BaseModel):
    invite_code: str = Field(min_length=9, max_length=9)


class OrgOut(ORMModel):
    id: uuid.UUID
    name: str
    address: str | None
    city: str | None
    state: str | None
    practice_type: str | None
    invite_code: str
    status: OrgStatus
    review_notes: str | None = None
    verified_at: datetime | None
    created_at: datetime
    member_count: int = 0


class OrgApproveIn(BaseModel):
    notes: str | None = Field(default=None, max_length=500)


class OrgRejectIn(BaseModel):
    reason: str = Field(min_length=1, max_length=500)


class OrgNetworkingSettingsIn(BaseModel):
    """Partial update — only supplied fields change (admin kill switch + policy)."""

    external_networking_enabled: bool | None = None
    external_dm_policy: ExternalDmPolicy | None = None
    directory_visibility: DirectoryVisibility | None = None


class OrgNetworkingSettingsOut(ORMModel):
    external_networking_enabled: bool
    external_dm_policy: ExternalDmPolicy
    directory_visibility: DirectoryVisibility


class MemberOut(BaseModel):
    """Minimum-necessary member view (HIPAA H8) — deliberately NO phone,
    email, or NPI. Full identifiers stay on /users/me (UserOut) only."""

    id: uuid.UUID
    full_name: str
    specialty: str | None
    org_role: OrgRole
    joined_at: datetime
    presence: str | None = None
    avatar_color: str | None = None
    avatar_url: str | None = None
    avatar_presigned_url: str | None = None
